pkg/build/builder/cmd/builder.go

```diff
@@ -67,6 +77,59 @@ func run(builderFactory factoryFunc) {
 
 }
 
+func setupSourceSecret(sourceSecretName string) (err error) {
+	// TODO: remove this when permissions for secrets will be fixed upstream
+	// this is to restrict access permissions to very low acceptable level
+	secretTmpDir, err := ioutil.TempDir("", "tmpsecret")
+	if err != nil {
+		return
+	}
+	cmd := exec.Command("cp", "-R", ".", secretTmpDir)
+	cmd.Dir = os.Getenv("SOURCE_SECRET_PATH")
+	if err = cmd.Run(); err != nil {
+		return
+	}
+	secretFiles, err := ioutil.ReadDir(secretTmpDir)
+	if err != nil {
+		return
+	}
+	for _, file := range secretFiles {
+		if err = os.Chmod(filepath.Join(secretTmpDir, file.Name()), 0600); err != nil {
+			return
+		}
+	}
+	os.Setenv("SOURCE_SECRET_PATH", secretTmpDir)
+	// TODO: end
+
+	sourceSecretDir := os.Getenv("SOURCE_SECRET_PATH")
```

Comment:
also move the 'fixing' part into `fixSecretPermissions()` method and add note about removing that once the problem is fixed in upstream.